### ISSUE TITLE
Bugs Releated To Tint2

### ISSUE THREAD

**Issue by Geyup** [bbidulock/blackboxwm#2]
Hi,

```My System Specs;```
Slackware 14.1 x86_64
Blackbox 0.71.1
tint2 0.12.7

Tint2 is back in active development and I hope you can please improve support for it.

https://gitlab.com/o9000/tint2/

I opened a ticket asking for support for the original Blackbox;

https://gitlab.com/o9000/tint2/issues/551

According to the developer his words;

**
Tint2 does not use custom code for every WM that is out there; instead, tint2 assumes that the WM implements a specification called EWMH, also called NetWM. This describes how the window manager interacts with other applications, such as the panel.

A lot of WMs implement this specification to the letter. Openbox, Fluxbox, Kwin, XFCE, GNOME and probably others. As a result tint2 works perfectly fine with all of these without needing any custom code.
**

This report I opened was with the original Blackbox 0.70.1, but since you've forked and updated I see one improvement, the ```numbers``` now appear in the tint2 panel. :)

These are the few issues I still have.

```1.``` I had to change the option ```disable_transparency = 1``` to use pseudo transparency. Using ```= 0``` doesn't work for true transparency, can Blackbox work with true transparency?

```2.``` Tint2 with the default setting of, ```panel_margin = 0 0``` will not allow tint2 to sit at the bottom of the screen, this setting works in Openbox. I had to change it to, ```panel_margin = 0 -33``` to push tint2 to the bottom of the screen. The tint2 developer suggested to see how ```screen margins``` are configured in blackbox.

```3``` Releated to what appears maybe ```screen margins``` again, when I maximize the size of a window, it fills the entire screen and goes behind tint2, it does not sit on the top of tint2 like it should.

```4.``` If I have a window minimized to tint2, and then click on another workspace section of tint2, the application follows and jumps over to the next workspace too, but if the window is not minimized, this doesn't happen.

**Comment by Geyup** [comment]
The argb-visual patch I mentioned in the Patch issue, also fixed ```transparency ```. I can  now use ```true transparency``` in blackbox for tint2 with that setting; ```disable_transparency = 0```

Please be sure to really look that patch over to see if it could use any changes.

**Comment by bbidulock** [comment]
The argb-visual.patch has been applied.

Support for _NET_WORKAREA is correct, I believe.  I don't know what you mean by screen margins.  _NET_REQUEST_FRAME_EXTENTS is not supported.  Neither is _NET_RESTACK_WINDOWS.

**Comment by Geyup** [comment]
I was told by the tint2 developer that since I need to adjust the option in the tint2 config ```panel_margin = 0 0``` that the problem might be related to the screen margins, how they are being used in Blackbox.

One thing I noticed, how the toolbar height in Blackbox, appears to be exactly the same size that tint2 will appear up off the bottom, like Blackbox has the bottom of the screen padded...

Numbers 2, 3, 4 are still problems with tint2 in Blackbox.

**Comment by bbidulock** [comment]
I don't see any problems with the behaviour of tint2 version 0.12 with regard to blackbox.

**Comment by Geyup** [comment]
What version of 0.12 are you using? I'm using the latest 0.12.7

You mentioned you added in the ```argb-visual.patch``` patch, yet I still see the same problems. And when I looked at the patch I see that you didn't add everything in, so something is missing that is effecting it for me.

Also conky doesn't show transparency too.

If I apply the patch in 0.70.0 original blackbox, or 0.71.1 everything looks good...

**Comment by Geyup** [comment]
I replied here; (transparency and workspace numbers working again)

https://github.com/bbidulock/blackboxwm/issues/7

I still have these issues I hope you can work out;

```2.``` Tint2 with the default setting of, panel_margin = 0 0 will not allow tint2 to sit at the bottom of the screen, this setting works in Openbox. I had to change it to, panel_margin = 0 -33 to push tint2 to the bottom of the screen. The tint2 developer suggested to see how screen margins are configured in blackbox.

```3.``` Releated to what appears maybe screen margins again, when I maximize the size of a window, it fills the entire screen and goes behind tint2, it does not sit on the top of tint2 like it should.

```4.``` If I have a window minimized to tint2, and then click on another workspace section of tint2, the application follows and jumps over to the next workspace too, but if the window is not minimized, this doesn't happen.

**Comment by Geyup** [comment]
I tried out Fluxbox 1.3.7 and I'm not seeing any of these bugs, so it has whatever support is needed, lacking in blackbox.

If it's the lack of _NET_REQUEST_FRAME_EXTENTS and _NET_RESTACK_WINDOWS, I hope you can please add this in later, or fix whatever is needed.

thanks

**Comment by Geyup** [comment]
@bbidulock, hi, have you had a chance to look at this?

thank you

**Comment by Geyup** [comment]
@bbidulock I guess you're one busy person... :)

tint2 does not work properly with Blackbox, I really hope you'll have time soon to fix this.

**Comment by bbidulock** [comment]
I did take a look at it.  I believe it is tint2's behavior that is at fault.  Much of what it does is a hack at best.  Rather than conforming to standards it appears to want to apply heursitics and execptions by window manager.  Use of a proper toolkit would yeild better results.  One thing that I found it doing was attempting to use static window gravity (and not checking when the window manager rightly sets it to NW gravity per ICCCM 2.0 specifications).  Correcting this in tint2 would like resolve positioning problems.  I looked at correcting this myself in tint2, but, well, I don't thinkt that I would do that because it would make me want to completely rewrite tint2.

I think that it is clear (at least to me) that the issues are with tint2 and not blackbox.
